packages/itwinui-react/src/core/Menu/MenuItem.tsx

```diff
@@ -179,7 +179,13 @@ export const MenuItem = React.forwardRef((props, forwardedRef) => {
   const popover = usePopover({
     nodeId,
     visible: isSubmenuVisible,
-    onVisibleChange: setIsSubmenuVisible,
+    onVisibleChange: (v) => {
+      if (!v) {
+        setHasFocusedNodeInSubmenu(false);
+      }
+
+      setIsSubmenuVisible(v);
+    },
```

Comment:
memoize + descriptive variable name

```suggestion
    onVisibleChange: React.useCallback((visible) => {
      if (!visible) {
        setHasFocusedNodeInSubmenu(false);
      }

      setIsSubmenuVisible(visible);
    }, []),
```

Reply:
Sounds good, done (https://github.com/iTwin/iTwinUI/pull/2026/commits/cf8b329c70f642100a517d94cb241ad3ed810b4b).